server/util/healthcheck/healthcheck.go

```diff
@@ -109,6 +112,41 @@ func (h *HealthChecker) Statusz(ctx context.Context) string {
 	return buf
 }
 
+func (h *HealthChecker) handleSignals(signalChan <-chan os.Signal) {
+	// When running in a terminal, the ^C character echoed back to the user
+	// messes up the log output a bit. So, print a newline every time we get
+	// a signal to make the output a little cleaner.
+	isTTY := isatty.IsTerminal(uintptr(os.Stderr.Fd()))
+	sig := <-signalChan
+	if isTTY {
+		fmt.Println()
+	}
+	log.Infof("Caught %s signal; starting graceful shutdown (hard-stopping in %s)", sig, *maxShutdownDuration)
+	hardStopTime := time.Now().Add(*maxShutdownDuration)
+	h.Shutdown()
+	// If we keep getting SIGINT/SIGTERM, the user is probably mashing
+	// Ctrl+C and really wants to kill the server. After 3 signals, exit
+	// immediately. Before then, report the current status so the user knows
+	// we got their request but are just still shutting down.
+	n := 0
+	const ignoreCount = 3
```

Comment:
I think this should be 2 if you want to match bazel's behavior? 1 for the initial shutdown and 2 more for frustration presses before you hard exit

Reply:
```
FAILED:
^C
Bazel caught interrupt signal; cancelling pending invocation.

^C
Bazel caught third interrupt signal; server killed. (This may be expensive, see https://bazel.build/advanced/performance/iteration-speed#avoid-ctrl-c.)

tylerw@lunchbox:~/buildbuddy$
```

Reply:
Done